<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from dobot.xacro.xml                | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="dobot">
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.028"/>
      <geometry>
        <box size="0.158 0.158 0.056"/>
      </geometry>
      <material name="White">
        <color rgba="1.0 1.0 1.0 1.0"/>
      </material>
    </visual>
  </link>
  <link name="link1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.041"/>
      <geometry>
        <box size="0.079 0.079 0.082"/>
      </geometry>
      <material name="White">
        <color rgba="1.0 1.0 1.0 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin rpy="0 0 0" xyz="0 0 0.056"/>
    <axis xyz="0 0 1"/>
    <limit effort="10" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="0.1"/>
  </joint>
  <link name="link2">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0675"/>
      <geometry>
        <box size="0.04 0.04 0.135"/>
      </geometry>
      <material name="White">
        <color rgba="1.0 1.0 1.0 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="0 0 0" xyz="0 0 0.082"/>
    <axis xyz="0 1 0"/>
    <limit effort="10" lower="0" upper="1.0471975511965976" velocity="0.1"/>
  </joint>
  <link name="link3">
    <visual>
      <origin rpy="0 1.5707963267948966 0" xyz="0.0725 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.145"/>
      </geometry>
      <material name="White">
        <color rgba="1.0 1.0 1.0 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0 0 0" xyz="0 0 0.135"/>
    <axis xyz="0 1 0"/>
    <limit effort="10" lower="-0.9599310885968813" upper="1.2217304763960306" velocity="0.1"/>
  </joint>
  <link name="link_mimic_1"/>
  <joint name="joint_mimic_1" type="revolute">
    <parent link="link3"/>
    <child link="link_mimic_1"/>
    <origin rpy="0 0 0" xyz="0.145 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="10" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.1"/>
    <mimic joint="joint2" multiplier="-1"/>
  </joint>
  <link name="link_mimic_2">
    <visual>
      <origin rpy="0 1.5707963267948966 0" xyz="0.03 0 0"/>
      <geometry>
        <box size="0.04 0.04 0.06"/>
      </geometry>
      <material name="White">
        <color rgba="1.0 1.0 1.0 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="joint_mimic_2" type="revolute">
    <parent link="link_mimic_1"/>
    <child link="link_mimic_2"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="10" lower="-3.141592653589793" upper="3.141592653589793" velocity="0.1"/>
    <mimic joint="joint3" multiplier="-1"/>
  </joint>
  <link name="link4">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.035"/>
      <geometry>
        <box size="0.02 0.02 0.07"/>
      </geometry>
      <material name="Red">
        <color rgba="1.0 0.0 0.0 1.0"/>
      </material>
    </visual>
  </link>
  <joint name="joint4" type="revolute">
    <parent link="link_mimic_2"/>
    <child link="link4"/>
    <origin rpy="0 0 0" xyz="0.06 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="10" lower="-2.6179938779914944" upper="2.6179938779914944" velocity="0.1"/>
  </joint>
  <link name="link_ee"/>
  <joint name="joint_ee" type="fixed">
    <parent link="link4"/>
    <child link="link_ee"/>
    <origin rpy="0 0 0" xyz="0 0 -0.07"/>
  </joint>
</robot>
